org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/LSPCodeActionMarkerResolution.java

```diff
@@ -147,31 +161,74 @@ private void checkMarkerResoultion(IMarker marker) throws IOException, CoreExcep
-					params.setRange(diagnostic.getRange());
-					CompletableFuture<List<Either<Command, CodeAction>>> codeAction = lsf
-							.thenComposeAsync(ls -> ls.getTextDocumentService().codeAction(params));
-					futures.add(codeAction);
-					codeAction.thenAcceptAsync(actions -> {
-						try {
-							marker.setAttribute(LSP_REMEDIATION, actions);
-						} catch (CoreException e) {
-							LanguageServerPlugin.logError(e);
+				final IEditorPart editor = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor();
+				if (editor instanceof ITextEditor) {
+					try {
+						Diagnostic diagnostic = (Diagnostic)marker.getAttribute(LSPDiagnosticsToMarkers.LSP_DIAGNOSTIC);
+						final ITextViewer textViewer = ((ITextEditor) editor).getAdapter(ITextViewer.class);
+						if (textViewer != null) {
+							for (CompletableFuture<LanguageServer> lsf : languageServerFutures) {
+								marker.setAttribute(LSP_REMEDIATION, COMPUTING);
+								CodeActionContext context = new CodeActionContext(Collections.singletonList(diagnostic));
+								CodeActionParams params = new CodeActionParams();
+								params.setContext(context);
+								params.setTextDocument(new TextDocumentIdentifier(LSPEclipseUtils.toUri(marker.getResource()).toString()));
+								params.setRange(diagnostic.getRange());
+								CompletableFuture<List<Either<Command, CodeAction>>> codeAction = lsf
+										.thenComposeAsync(ls -> ls.getTextDocumentService().codeAction(params));
+								futures.add(codeAction);
+								codeAction.thenAcceptAsync(actions -> {
+									try {
+										marker.setAttribute(LSP_REMEDIATION, actions);
+										PlatformUI.getWorkbench().getDisplay().asyncExec(() -> reinvokeQuickfixProposalsIfNecessary(textViewer));
+									} catch (CoreException e) {
+										LanguageServerPlugin.logError(e);
+									}
+								});
+							}
 						}
-					});
+					} catch (Exception e) {
+						LanguageServerPlugin.logError(e);
+					}
 				}
+
 				// wait a bit to avoid showing too much "Computing" without looking like a freeze
 				CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()])).get(300, TimeUnit.MILLISECONDS);
 			}
 		}
 	}
 
+	private void reinvokeQuickfixProposalsIfNecessary(ITextViewer textViewer) {
+		try {
+			// Quick assist proposals popup case
+			if (textViewer instanceof ISourceViewerExtension3) {
+				IQuickAssistAssistant quickAssistant = ((ISourceViewerExtension3)textViewer).getQuickAssistAssistant();
+					Field f = QuickAssistAssistant.class.getDeclaredField("fQuickAssistAssistantImpl"); //$NON-NLS-1$
+					f.setAccessible(true);
+					ContentAssistant ca = (ContentAssistant) f.get(quickAssistant);
+					Method m = ContentAssistant.class.getDeclaredMethod("isProposalPopupActive"); //$NON-NLS-1$
+					m.setAccessible(true);
+					boolean isProposalPopupActive = (Boolean) m.invoke(ca);
```

Comment:
I think we can leverage the (internal) implementation detail that `QuickAssistAssistantImpl extends ContentAssistant` and ContentAssistant allows some listeners that might be useful to deduce whether the dialog is active or not.

Reply:
Another possibility is to also check the current Shell and find a Table child and verify whether it does contain the usual "Computing..." item so we know we can replace it.

Reply:
Tried adding a `ICompletionListener` to quick assistant. Didn't work. The `LSPCodeActionMarkerResolution` is created when the completion proposal session is started already... This results in indeterminate state for quick assistant proposals being active or not because it is unclear what was the starting state. If no session start/end event received it is unclear if quick assist proposals are active or not.
The Shell-Table Computing entry in the table solution would work i think... However, I feel it'd be less hacky to add `boolean isProposalPopupActive()` to `IQuickAssistant` interface. If you think it is a good option, I'd create a PR for Eclipse Platform Text.